### PR TITLE
Fix Maven cache move

### PR DESCRIPTION
**Pull request by SohamDas2021**
## Summary of the pull request
Dev drive documentation initially suggested moving all contents of %userprofile%/.m2 to dev drive for optimization. However we only need to move contents of %userprofile%/.m2/repository to dev drive and set MAVEN_OPTS="-Dmaven.repo.local=<dev drive location for .m2\repository>"

## References and relevant issues
#3336 

## Validation steps performed
Manual tests. Tested with real Maven cache and repository.

### CODE REVIEW

File: tools/Customization/DevHome.Customization/ViewModels/DevDriveInsights/OptimizeDevDriveDialogViewModel.cs [hunk 1]
```diff
@@ -224,6 +224,14 @@ private void SetEnvironmentVariable(string variableName, string value)
     {
         try
         {
+            if (string.Equals(variableName, "MAVEN_OPTS", StringComparison.OrdinalIgnoreCase))
+            {
+                var existingValue = Environment.GetEnvironmentVariable(variableName, EnvironmentVariableTarget.User);
+                var newValue = (!string.IsNullOrEmpty(existingValue) ? existingValue + " " : string.Empty) + "-Dmaven.repo.local = " + value;
```
Comment (dhoehna) [inline tools/Customization/DevHome.Customization/ViewModels/DevDriveInsights/OptimizeDevDriveDialogViewModel.cs:230]: NIT: Consider making `-Dmaven.repo.local =` a const string in a common location.

**Review by dhoehna — APPROVED** [review]

**Review by kanismohammed — APPROVED** [review]

File: tools/Customization/DevHome.Customization/ViewModels/DevDriveInsights/OptimizeDevDriveDialogViewModel.cs [hunk 2]
```diff
@@ -220,10 +223,86 @@ private void UpdatePathEnvironmentVariable(string value)
...
+            }
+        }
+
+        doc.Save(xmlFilePath);
+    }
+
+    private void UpdateSettingsXML(string localRepositoryLocation)
+    {
+        int index = ExistingCacheLocation.IndexOf("repository", StringComparison.OrdinalIgnoreCase);
+        var settingsXMLPath = string.Concat(this.ExistingCacheLocation.AsSpan(0, index), "settings.xml");
```
Comment (dhoehna) [inline tools/Customization/DevHome.Customization/ViewModels/DevDriveInsights/OptimizeDevDriveDialogViewModel.cs:267]: remove `this`

File: tools/Customization/DevHome.Customization/ViewModels/DevDriveInsights/OptimizeDevDriveDialogViewModel.cs [hunk 3]
```diff
@@ -220,10 +223,86 @@ private void UpdatePathEnvironmentVariable(string value)
...
+                doc.DocumentElement.AppendChild(newKey);
+            }
+        }
+
+        doc.Save(xmlFilePath);
+    }
+
+    private void UpdateSettingsXML(string localRepositoryLocation)
+    {
+        int index = ExistingCacheLocation.IndexOf("repository", StringComparison.OrdinalIgnoreCase);
```
Comment (dhoehna) [inline tools/Customization/DevHome.Customization/ViewModels/DevDriveInsights/OptimizeDevDriveDialogViewModel.cs:266]: What if -1 is returned?

File: tools/Customization/DevHome.Customization/ViewModels/DevDriveInsights/OptimizeDevDriveDialogViewModel.cs [hunk 4]
```diff
@@ -220,10 +223,86 @@ private void UpdatePathEnvironmentVariable(string value)
...
+        var settingsXMLPath = string.Concat(this.ExistingCacheLocation.AsSpan(0, index), "settings.xml");
+        var key = "settings";
+        var subkey = "localRepository";
+
+        if (File.Exists(settingsXMLPath))
+        {
+            // Update settings.xml with the new repository path
+            UpdateOrCreateSubkey(settingsXMLPath, key, subkey, localRepositoryLocation);
```
Comment (dhoehna) [inline tools/Customization/DevHome.Customization/ViewModels/DevDriveInsights/OptimizeDevDriveDialogViewModel.cs:274]: Comment says `Update...` but the method name is `UpdateOrCreate` please update the comment.

Comment (SohamDas2021) [reply tools/Customization/DevHome.Customization/ViewModels/DevDriveInsights/OptimizeDevDriveDialogViewModel.cs:274]: It updates or creates the subkey, not the file itself.

File: tools/Customization/DevHome.Customization/ViewModels/DevDriveInsights/OptimizeDevDriveDialogViewModel.cs (hunk 3 above)
Comment (SohamDas2021) [reply tools/Customization/DevHome.Customization/ViewModels/DevDriveInsights/OptimizeDevDriveDialogViewModel.cs:266]: This code won't execute if index is -1 here. So we are good.

**Pull request merged**